### PR TITLE
Mass delete courses

### PR DESCRIPTION
**Pull request by ltwheeler**
Adds the ability to mass delete courses

### CODE REVIEW

File: src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java [hunk 1]
```diff
@@ -146,6 +146,24 @@ public void deleteCourse(@PathVariable long courseId, HttpServletResponse httpRe
 		courseService.delete(courseId);
 	}
 
+	@RequestMapping(value = "/api/courseView/schedules/{workgroupId}/{year}/courses", method = RequestMethod.PUT, produces="application/json")
+	@ResponseBody
+	public List<Long> deleteMultipleCourses(@PathVariable long workgroupId, @PathVariable long year, @RequestBody List<Long> courseIds, HttpServletResponse httpResponse) {
+		Schedule schedule = scheduleService.findByWorkgroupIdAndYear(workgroupId, year);
```
Comment (cthielen) [inline src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java]: Does Spring allow workgroupId or year to be null?

File: src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java [hunk 2]
```diff
@@ -146,6 +146,24 @@ public void deleteCourse(@PathVariable long courseId, HttpServletResponse httpRe
...
+
+		if (schedule == null) {
+			httpResponse.setStatus(HttpStatus.BAD_REQUEST.value());
```
Comment (cthielen) [inline src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java]: 404 is more appropriate. Bad request would be if the request isn't understood, while 404 means the request is understood (e.g. a workgroupId and year were given) but that there's no data to be found.

File: src/main/java/edu/ucdavis/dss/ipa/repositories/CourseRepository.java [hunk 3]
```diff
@@ -62,4 +63,13 @@ Course findBySubjectCodeAndCourseNumberAndSequencePatternAndEffectiveTermCodeAnd
     List<Course> findBySubjectCodeAndCourseNumberAndSequencePatternAndScheduleId(String subjectCode, String courseNumber, String sequencePattern, long scheduleId);
 
     Course findOneBySubjectCodeAndCourseNumberAndSequencePatternAndScheduleId(String subjectCode, String courseNumber, String sequencePattern, long scheduleId);
+
+    /**
+     * Delete all Courses with ids specified in {@code ids} parameter
+     *
+     * @param courseIds List of user ids
+     */
+    @Modifying
```
Comment (cthielen) [inline src/main/java/edu/ucdavis/dss/ipa/repositories/CourseRepository.java]: Neat!

File: src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java (hunk 2 above)
Comment (ltwheeler) [reply src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java]: Hmmmm, you are correct. 404 feels odd, but it is definitely more appropriate to this situation of 'target resource not found'

File: src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java (hunk 1 above)
Comment (ltwheeler) [reply src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java]: Spring will throw a 400 if either are null.

**Pull request merged**